build-system/babel-plugins/babel-plugin-dom-jsx-svg-namespace/test/fixtures/transform/sets-prop/output.mjs

```diff
@@ -0,0 +1,26 @@
+/** @jsx jsx.createElement */
+import * as jsx from '../ANYWHERE_THAT_LEADS_TO/core/dom/jsx';
+
+() => jsx.createElement("svg", {
+  __svg: true
+});
+
+() => jsx.createElement("svg", {
+  __svg: true
+});
+
+() => jsx.createElement("path", {
+  foo: "bar",
+  __svg: true
```

Comment:
The problem lies in that the calls to create the child `path` are executed first, at which point we may create it without a namespace. It's also detached so it has no context in the tree:

```js
createElement('svg', null, createElement('circle'))
```

We would have the same issue with an annotating function, since the element would have been created by the time it's passed as an argument:

```js
svgTree(createElement('circle'))
```

Alternatively, we could verify that a svg subtree is static, and transform it into an `svgFor` call. But that may be more trouble than it's worth.

Reply:
Another alternative is to refactor the renderer so that the nodes are immediate representations, and modify this transform so that the root is concretized:

```
// input:
<foo><svg><path /></svg></foo>;

// ouptut:
concretize(createElement('foo', null, createElement('svg', null, createElement('bar'))));

// during execution:
concretize(['foo', /* svg */ false, null, ['svg', true, null, ['path', true]]]);
```

Reply:
Sorry, my comment on `<svg><circle /></svg>` working correctly was with that assumption that we land this PR (but in a changed state where only static descendants of an `<svg>` node are marked with `__svg`).

With `concretize`, the same issue would still appear with a root `<circle />` node, where it would currently be created in the SVG namespace where it used to be created in the HTML namespace with `htmlFor`. If we had something like `let child = <circle />; <foo>{child}</foo>`, we'd still have the incorrectly created circle node, because it doesn't exist in an SVG. But something like `<svg>{child}</svg>` would now be supported easily, so that's a good change.

My suggestion with `svgTree` helper was so that we can syntaticaly mark partial SVG trees, and now embedding inside an HTML without an parent `<svg>` would work: `let child = svgTree(<circle />); <foo>{child}</foo>` would do something like `svgConcretize(createElement('circle'))`.